Source/WebCore/Modules/mediastream/ImageCapture.cpp

```diff
@@ -79,16 +79,17 @@ void ImageCapture::takePhoto(PhotoSettings&& settings, DOMPromiseDeferred<IDLInt
         return;
     }
 
-    m_track->takePhoto(WTFMove(settings))->whenSettled(RunLoop::protectedMain(), [this, protectedThis = Ref { *this }, promise = WTFMove(promise), identifier = WTFMove(identifier)] (auto&& result) mutable {
-        legacyQueueTaskKeepingObjectAlive(*this, TaskSource::ImageCapture, [this, promise = WTFMove(promise), result = WTFMove(result), identifier = WTFMove(identifier)] () mutable {
+    Ref protectedThis = Ref { *this };
+    m_track->takePhoto(WTFMove(settings))->whenSettled(RunLoop::protectedMain(), [protectedThis = WTFMove(protectedThis), promise = WTFMove(promise), identifier = WTFMove(identifier)] (auto&& result) mutable {
+        legacyQueueTaskKeepingObjectAlive(protectedThis.get(), TaskSource::ImageCapture, [protectedThis = WTFMove(protectedThis), promise = WTFMove(promise), result = WTFMove(result), identifier = WTFMove(identifier)] () mutable {
```

Comment:
`protectedThis` is moved in the lambda capture, so `protectedThis.get()` is accessing the Ref after it's been moved and is now null.